### ISSUE TITLE
ReplaceSafeCallChainWithRun – false positive on platform types

### ISSUE THREAD

**Issue by zarebski-m** [detekt/detekt#7849]
## Expected Behavior

Consider the following code:

```kotlin
import io.vertx.core.json.JsonObject
import io.vertx.ext.web.client.HttpResponse

fun foo(response: HttpResponse<JsonObject>): String = 
    response.body()
        .getJsonArray("value")
        ?.takeUnless(JsonArray::isEmpty)
        ?.getJsonObject(0)
        ?.getString("id")
```

This code should not trigger `ReplaceSafeCallChainWithRun` because all of the methods in the chain return either a nullable or platform type. Replacing it with `run` would change the behavior and might lead to NPE at runtime.

## Observed Behavior

`ReplaceSafeCallChainWithRun` is reported.

## Your Environment

* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): N/A, used with Maven and `com.github.ozsie:detekt-maven-plugin:1.23.7`
* Version of Kotlin used: 2.0.0
* Version of Java used: Temurin 21.0.5

**Comment by BraisGabin** [comment]
True, platform types should be considered as nullable in this rule.

**Comment by atulgpt** [comment]
Hi @zarebski-m can you try on snapshot release, I tried to add TC on https://github.com/detekt/detekt/pull/8195 around platform types but it was passing

**Comment by BraisGabin** [comment]
Closing as this seems to be fixed already.
